src/utils/ProtectedRoute.jsx

```diff
@@ -1,14 +1,18 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Navigate } from 'react-router-dom';
 import { PropTypes, instanceOf } from 'prop-types';
-
 import { withCookies, cookieKeys, Cookies } from './cookie_utils';
-import { auth, getCurrentUser } from './auth_utils';
+import { NPOBackend, refreshToken } from './auth_utils';
 
 // TODO: Make calls to backend to verify user access token
-// const verifyToken = () => {
-//   return true;
-// };
+const userIsAuthenticated = async (roles, cookies) => {
+  const accessToken = await refreshToken(cookies);
```

Comment:
prob not the best idea to refresh the token every time protected routes is called. `userIsAuthenticated` should only check the current cookies. since we refresh the token when there is an error with the backend, zubair and i think its fine. but if this doesnt work, feel free to leave it as it is.